RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -118,63 +169,114 @@ class EvtPlaneProducer : public edm::EDProducer {
+
+  edm::InputTag castorTag_;
+  edm::EDGetTokenT<std::vector<reco::CastorTower>> castorToken;
+  edm::Handle<std::vector<reco::CastorTower>> castorCollection_;
+
+  edm::InputTag trackTag_;
+  edm::EDGetTokenT<reco::TrackCollection> trackToken;
+  edm::Handle<reco::TrackCollection> trackCollection_;
+
+  bool foundCentTag;
   bool useECAL_;
   bool useHCAL_;
   bool useTrack_;
-  bool useTrackPtWeight_;
+  bool loadDB_;
   double minet_;
   double maxet_;
+  double effm_;
   double minpt_;
   double maxpt_;
   double minvtx_;
   double maxvtx_;
   double dzerr_;
   double chi2_;
-
-  bool storeNames_;
+  bool FirstEvent;
+  int FlatOrder_;
+  uint runno_; 
+  int NumFlatBins_;
+  double nCentBins_;
+  double caloCentRef_;
+  double caloCentRefWidth_;
+  int caloCentRefMinBin_;
+  int caloCentRefMaxBin_;
+  int CentBinCompression_;
+  HiEvtPlaneFlatten * flat[NumEPNames];
 };
 
-//
-// constants, enums and typedefs
-//
+EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet& iConfig) {
 
+  nCentBins_ = 200.;
 
-//
-// static data member definitions
-//
+  centralityVariable_ = iConfig.getParameter<std::string>("centralityVariable");
+  if(iConfig.exists("nonDefaultGlauberModel")){
+    centralityMC_ = iConfig.getParameter<std::string>("nonDefaultGlauberModel");
+  }
+  centralityLabel_ = centralityVariable_+centralityMC_;
 
-//
-// constructors and destructor
-//
-EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet& iConfig)
-{
-  
-  
-  //register your products
-  vtxCollection_    = consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vtxCollection_"));
-  caloCollection_   = consumes<CaloTowerCollection>(iConfig.getParameter<edm::InputTag>("caloCollection_"));
-  trackCollection_  = consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("trackCollection_"));
-  useECAL_ = iConfig.getUntrackedParameter<bool>("useECAL_",true);
-  useHCAL_ = iConfig.getUntrackedParameter<bool>("useHCAL_",true);
-  useTrack_ = iConfig.getUntrackedParameter<bool>("useTrack",true);
-  useTrackPtWeight_ = iConfig.getUntrackedParameter<bool>("useTrackPtWeight_",true);
-  minet_ = iConfig.getUntrackedParameter<double>("minet_",0.2);
-  maxet_ = iConfig.getUntrackedParameter<double>("maxet_",500.);
+  centralityBinTag_ = iConfig.getParameter<edm::InputTag>("centralityBinTag_");
+  centralityBinToken = consumes<int>(centralityBinTag_);
+
+  vertexTag_  = iConfig.getParameter<edm::InputTag>("vertexTag_");
+  vertexToken = consumes<std::vector<reco::Vertex>>(vertexTag_);
+
+  caloTag_ = iConfig.getParameter<edm::InputTag>("caloTag_");
+  caloToken = consumes<CaloTowerCollection>(caloTag_);
+
+  castorTag_ = iConfig.getParameter<edm::InputTag>("castorTag_");
+  castorToken = consumes<std::vector<reco::CastorTower>>(castorTag_);
+
+  trackTag_ = iConfig.getParameter<edm::InputTag>("trackTag_");
+  trackToken = consumes<reco::TrackCollection>(trackTag_);
+
+  FlatOrder_ = iConfig.getUntrackedParameter<int>("FlatOrder_", 9);
```

Comment:
as mentioned in config file comments: this and many other parameters should be tracked (remove "Untracked")